### ISSUE TITLE
tracing/eventsource/eventsourcetrace/eventsourcetrace/eventsourcetrace.sh failure

### ISSUE THREAD

**Issue by mikedn** [dotnet/coreclr#20510]
In Ubuntu16.04 arm64 Cross Checked Innerloop Build and Test this test failed with the following error:
```
Assert failure(PID 32988 [0x000080dc], Thread: 32988 [0x80dc]): Consistency check failed: Crst Level violation: Can't take level 18 lock CrstPendingTypeLoadEntry because you already holding level 17 lock CrstEventPipe
```
in https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm64_cross_checked_ubuntu16.04_innerloop_flow_prtest/545/ (triggered in #20045).

I think I've already seen some issue(s) about tracing but I can't seem to find anything that involves crst level violation so opened a new issue.

**Comment by RussKeldorph** [comment]
@brianrob

**Comment by noahfalk** [comment]
Thanks for the report @mikedn ! Let us know if there is anything pressing here, otherwise it might be a while before we can get to this. We just inherited the area from Brian and there has been quite an influx new issues that we aren't yet ramped up to handle quickly.

**Comment by mikedn** [comment]
Seems like a rare issue, at least in comparation to other issues that hit CI rather often. I've yet to see this again.

**Comment by jkotas** [comment]
Hit by: https://github.com/dotnet/coreclr/pull/20878

**Comment by jkotas** [comment]
Hit again by https://github.com/dotnet/coreclr/pull/21071 (https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm64_cross_checked_ubuntu16.04_innerloop_flow_prtest/1787/)

**Comment by tommcdon** [comment]
cc @jorive

**Comment by tommcdon** [comment]
@sywhang has a WIP that should address this
Related to https://github.com/dotnet/coreclr/issues/23412

**Comment by sywhang** [comment]
#24101 should have addressed this issue. I opened a PR to re-enable this test: https://github.com/dotnet/coreclr/pull/24365.

**Comment by sywhang** [comment]
#24365 re-enabled this test. Please feel free to open this issue again if another failure is observed.
